v1/plugins/plugins.go

```diff
@@ -594,14 +593,29 @@ func (m *Manager) Init(ctx context.Context) error {
 func (m *Manager) Labels() map[string]string {
 	m.mtx.Lock()
 	defer m.mtx.Unlock()
-	return m.Config.Labels
+
+	return maps.Clone(m.Config.Labels)
 }
 
 // InterQueryBuiltinCacheConfig returns the configuration for the inter-query caches.
 func (m *Manager) InterQueryBuiltinCacheConfig() *cache.Config {
 	m.mtx.Lock()
 	defer m.mtx.Unlock()
-	return m.interQueryBuiltinCacheConfig
+
+	if m.interQueryBuiltinCacheConfig == nil {
+		return nil
+	}
+
+	clone := *m.interQueryBuiltinCacheConfig
```

Comment:
Is this enough considering that the fields of `InterQueryBuiltinCacheConfig` and `InterQueryBuiltinValueCacheConfig` are mostly pointers? Maybe we need to add `Clone()` methods to these (and `cache.Config`) too?